odoo/orm/models.py

```diff
@@ -5744,8 +5745,12 @@ def _update_cache(self, values: ValuesType, validate: bool = True) -> None:
         # convert monetary fields after other columns for correct value rounding
         for field, value in sorted(field_values, key=lambda item: item[0].write_sequence):
             value = field.convert_to_cache(value, self, validate)
-            field_cache = self.env._field_cache(field)
-            field_cache[self.id] = value
+            dirty = field.store and any(self._ids)
+            field._cache_update(self, value, dirty=dirty)
```

Comment:
The goal of `_update_cache()` is to update the cache _without making the field dirty_.  So this simply cannot be correct IMHO.

Reply:
Will remove this change.